controllers/managedocs_controller.go

```diff
@@ -255,3 +344,138 @@ func (r *ManagedOCSReconciler) updateComponents() v1.ComponentStatusMap {
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return false
+		}
+		r.Log.Error(err, "Unable to get configMap")
+		return false
+	}
+	labels := configmap.Labels
+	_, ok := labels[r.DeleteConfigMapLabelKey]
+	return ok
+}
+
+func (r *ManagedOCSReconciler) removeOlmComponents() error {
+
+	r.Log.Info("Deleting subscription")
+	subscription := &operators.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.AddonSubscriptionName,
+			Namespace: r.namespace,
+		},
+	}
+	if err := r.deleteResource(r.Client, subscription, r.AddonSubscriptionName); err != nil {
+		return err
+	}
+	r.Log.Info("subscription removed successfully")
+
+	r.Log.Info("Deleting CSV")
+	csvList := &operators.ClusterServiceVersionList{}
+	err := r.Client.List(r.ctx, csvList)
+	if err != nil {
+		r.Log.Error(err, "Unable to list CSVs")
+		return err
+	}
+	for _, csv := range csvList.Items {
+		if strings.HasPrefix(csv.Name, deployerCsvPrefix) {
+			deployerCsvName := csv.Name
+			csv := &operators.ClusterServiceVersion{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      deployerCsvName,
+					Namespace: r.namespace,
+				},
+			}
+			if err := r.deleteResource(r.Client, csv, deployerCsvName); err != nil {
+				return err
+			}
+		}
+	}
+	r.Log.Info("CSV removed successfully")
+
+	return nil
+}
+
+func (r *ManagedOCSReconciler) areComponentsReadyForUninstall() bool {
+	subComponent := r.managedOCS.Status.Components
+
+	if subComponent.StorageCluster.State != v1.ComponentReady {
+		return false
+	}
+	return true
+}
+
+func (r *ManagedOCSReconciler) deleteComponents() error {
+	r.Log.Info("Deleting storageCluster")
+	if err := r.deleteResource(r.Client, &ocsv1.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{Name: storageClusterName, Namespace: r.namespace},
+	}, storageClusterName); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *ManagedOCSReconciler) areComponentsDeleted() bool {
+	subComponent := r.managedOCS.Status.Components
+
+	if subComponent.StorageCluster.State == v1.ComponentNotFound {
+		return true
+	}
+	return false
+}
+
+func (r *ManagedOCSReconciler) findOcsPvcs() (bool, error) {
+	pvcList := &corev1.PersistentVolumeClaimList{}
+	err := r.UnrestrictedClient.List(r.ctx, pvcList)
+	if err != nil {
+		return false, fmt.Errorf("Unable to list PVCs: %v", err)
+	}
+	for _, pvc := range pvcList.Items {
+		scName := *pvc.Spec.StorageClassName
+		if scName == storageClassCephFsName || scName == storageClassRbdName {
+			r.Log.Info("Found Consumer PVCs using OCS storageclasses, cannot proceed on uninstalltion")
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (r *ManagedOCSReconciler) deleteResource(cl client.Client, resource runtime.Object, resourceName string) error {
+	err := cl.Delete(r.ctx, resource)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		r.Log.Error(err, "Uninstall: unable to delete resource", "resource", resourceName)
+		return err
+	}
+	return nil
+}
```

Comment:
Two observations:
1. you do not need to pass `resourceName` as it can be extracted from the passed resource.
2. I do not see a point for this function as we can just replace it (on the call site) with a single liner 
```
if err := r.client.delete(r.ctx, resource); err != nil && !errors.IsNotFound(err) {
```

Reply:
Single liner is precise and thus removing the usage of a function to perform deletion.